docs/reference/index.rst

```diff
@@ -0,0 +1,32 @@
+.. eql:section-intro-page:: reference
+
+.. _ref_reference_index:
+
+=========
+Reference
+=========
+
+.. toctree::
+    :maxdepth: 3
+    :hidden:
+
+    dev
+    lexical
+    eval
+    shapes
+    output
+    paths
+    observability
+    functions
+    dump_format
+    backend_ha
+    configuration
+    protocol/index
+    commands/index
```

Comment:
I think commands/sdl/ddl should be at the top of the section

Reply:
Relatedly, there are now several EdgeQL-related pages that probably desserve a separate subsection

Lexical Structure
Evaluation Algorithm
Shapes
Output Shaping
Paths
Function Calls

Some of these are pages I added to include a formal definition for non-command syntax like shapes and functions calls. Others are detailed guides of certain concepts. We can either place these in their own subsection or put them under "Commands" (and renaming "Commands" to "EdgeQL"). Thought?